### ISSUE TITLE
Script or local server interface

### ISSUE THREAD

**Issue by TheDuckCow** [jmc2obj/j-mc-2-obj#121]
This is a lofty ask, but throwing it out there nonetheless. Minewayas has a feature where you can [control the program via a script](http://www.realtimerendering.com/erich/minecraft/public/mineways/scripting.html), including steps like loading worlds, setting export settings, and even running exports (or multiple, with different specifications of volumes to export each time). 

For context, I'm the developer behind the blender [MCprep addon](https://github.com/TheDuckCow/MCprep). I saw the Mienways script interface as an opportunity to build a [dynamic "world importer bridge" in blender](https://twitter.com/TheDuckCow/status/1151577959284793344), without commiting to reinvent the world of making my own obj exporter/interface. This is still a work in progress, but functional nonetheless.

**So, the feature proposal:**

- Develop a local server API interface which will allow outside programs to talk to jmc2obj without direct human interaction
  - Or, if that's too difficult, then implementing a similar approach as Mineways which offers execution of a script file specified via command line when the program launches as an additional argument.
  - Issue with this however, is the program has to be instructed to quit after each individual export command, as once the script completes no further commands can be issued and thus the process would become orphaned.
- Have this API or script interface implement endpoints for applying settings
- Have this API or script interface trigger certain commands such as load world, export selected area or export textures
- Implement a headless mode so the UI is not drawn or ever opened, to be optimally fast and function as a backend for other programs to manage.

Again, I realize this is an ordeal to create (and, as a developer, I'm probably shooting myself in the foot by asking to have two systems to support), but I figure I should at least throw the idea out there for neutrality's sake. I'm happy to also share progress I make with this concept with Mineways if you are curious to follow, see the [outline here](https://github.com/TheDuckCow/MCprep/issues/75).

**Comment by paol** [comment]
I added the command line mode back in the day for just such automation needs. Run with --help to see the options.

**Comment by TheDuckCow** [comment]
Great, thanks! This seems to work, just tried it with an example. One thing I noticed though when exporting the textures from commandline, it seems to go incredibly slowly - where the GUI might take a second, the command line export took nearly 10 minutes (with lots of printout statements like the below)
```
Trying normal map
Error loading normal texture: Couldn't find assets/minecraft/textures/painting/sea_n.png in 1.14.3.jar
Trying specular map
Error loading specular image: Couldn't find assets/minecraft/textures/painting/sea_s.png in 1.14.3.jar
```

I wonder if the console printouts are actually what makes it slower, as I've experienced that before. Is there a way to reduce the verbosity of logging? Or, might there be another reason why texture exporting via command line is so slow?

Beyond this, the interface you have is perfect for what I need.
